lib/lightstep/tracer.rb

```diff
@@ -53,72 +53,61 @@ def report_period_seconds=(seconds)
       @reporter.period = seconds
     end
 
-    # TODO(ngauthier@gmail.com) inherit SpanContext from references
+    # TODO(bhs): Support FollowsFrom and multiple references
 
     # Starts a new span.
-    # @param operation_name [String] the operation name for the Span
-    # @param child_of [Span] Span to inherit from
+    #
+    # @param operation_name [String] The operation name for the Span
+    # @param child_of [SpanContext] SpanContext that acts as a parent to
+    #        the newly-started Span. If a Span instance is provided, its
+    #        .span_context is automatically substituted.
     # @param start_time [Time] When the Span started, if not now
-    # @param tags [Hash] tags for the span
+    # @param tags [Hash] Tags to assign to the Span at start time
     # @return [Span]
     def start_span(operation_name, child_of: nil, start_time: nil, tags: nil)
-      child_of_id = nil
-      trace_id = nil
-      if Span === child_of
-        child_of_id = child_of.span_context.id
-        trace_id = child_of.span_context.trace_id
-      else
-        trace_id = LightStep.guid
-      end
-
       span = Span.new(
         tracer: self,
         operation_name: operation_name,
-        child_of_id: child_of_id,
-        trace_id: trace_id,
+        child_of: child_of,
         start_micros: start_time.nil? ? LightStep.micros(Time.now) : LightStep.micros(start_time),
         tags: tags,
-        max_log_records: max_log_records
+        max_log_records: max_log_records,
       )
 
-      if Span === child_of
-        span.set_baggage(child_of.span_context.baggage)
-      end
-
       span
```

Comment:
Yeah, I know, formally... I have seen this pattern in some other ruby repos (where it's fashionable to have a local var sit on the last line to make return semantics consistent). YMMV? Ruby idioms are so baffling to me that I end up feeling like I'm doing cargo-cult stuff, like this.

Since you mentioned it, I will change this, though. We both can't be crazy. :)